src/Git/Index/Extensions/CachedTree.hs

```diff
@@ -0,0 +1,38 @@
+module Git.Index.Extensions.CachedTree where
+
+import Data.Attoparsec.ByteString.Char8 (decimal)
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Data.Word
+import qualified Data.Text as Text
+import qualified System.Path as Path
+
+import Git.Index.Extensions.Class (IndexExtension(..))
+import Git.Internal (takeFor, nullTermStringP, char_)
+import Git.Sha1 (Sha1, sha1HexParser)
+
+
+newtype CachedTree
+  = CachedTree (Map Path.RelFileDir CachedTreeRow) deriving (Show)
+
+data CachedTreeRow
+  = CachedTreeRow
+  { ctrEntryCount :: Word32
```

Comment:
How are we going to deal with -1 which indicates an entry in an invalidated state?

Reply:
I think my plan at the moment is that we handle that on parsing, and that it's not part of this data structure once the data has made it through the parse. _However_, I'm not entirely sure what each bit of this data structure is used for, so it might be that we have to rethink that when we start using it.

The bigger problem here, I would guess, is that we haven't accounted for the negative number in the parser yet?

Reply:
captured in issue #21